spec/features/reservations_spec.rb

```diff
@@ -376,4 +414,214 @@
       end
     end
   end
+
+  context 'valid items on confirmation page' do
+    before(:each) do
+      empty_cart
+      select('10', from: 'items_per_page')
+      find('#items_per_form').submit_form!
+      add_item_to_cart(@eq_model)
+      add_item_to_cart(@eq_model2)
+      update_cart_start_date(Time.zone.today)
+      due_date = Time.zone.today + 1.day
+      update_cart_due_date(due_date)
+      visit new_reservation_path
+    end
+
+    shared_examples 'can make valid change to reservation' do |reserver|
+      let(:reserver) { reserver }
+      str = reserver ? 'for other user successfully' : 'successfully'
+      it str do
+        quantity_forms = page.all('#quantity_form')
+        fill_in "quantity_field_#{@eq_model.id}", # edit and submit
+                with: (@eq_model.max_per_user)
```

Comment:
This line can result in failing validations if there aren't enough equipment items for that model (FactoryGirl defaults `max_per_user` to 10). You'd have to create additional equipment items to make sure there's enough availability.